### PR TITLE
Unset FEATURE_EVENTSOURCE_XPLAT in macOS builds

### LINKED ISSUE #48233: Regression causing new Apple Silicon failures
There are a 109 new tests failing on a recent tip.

The tests pass if `COMPlus_EnableEventLog` is not set, but the flag is set by default by the test run scripts.  (It can be disabled with `--disableEventLogging`)

```
baseservices/threading/generics/WaitCallback/thread01/thread01.sh 
baseservices/threading/generics/WaitCallback/thread02/thread02.sh 
baseservices/threading/generics/WaitCallback/thread03/thread03.sh 
baseservices/threading/generics/WaitCallback/thread04/thread04.sh 
baseservices/threading/generics/WaitCallback/thread05/thread05.sh 
baseservices/threading/generics/WaitCallback/thread06/thread06.sh 
baseservices/threading/generics/Monitor/EnterExit01/EnterExit01.sh             
baseservices/threading/generics/Monitor/EnterExit02/EnterExit02.sh             
baseservices/threading/generics/Monitor/EnterExit03/EnterExit03.sh             
baseservices/threading/generics/Monitor/EnterExit04/EnterExit04.sh             
baseservices/threading/generics/Monitor/EnterExit05/EnterExit05.sh             
baseservices/threading/generics/Monitor/EnterExit06/EnterExit06.sh             
baseservices/threading/generics/Monitor/EnterExit07/EnterExit07.sh             
baseservices/threading/generics/Monitor/EnterExit08/EnterExit08.sh             
baseservices/threading/generics/Monitor/EnterExit09/EnterExit09.sh             
baseservices/threading/generics/Monitor/EnterExit10/EnterExit10.sh             
baseservices/threading/generics/Monitor/EnterExit11/EnterExit11.sh             
baseservices/threading/generics/Monitor/EnterExit12/EnterExit12.sh             
baseservices/threading/generics/Monitor/EnterExit13/EnterExit13.sh             
baseservices/threading/generics/Monitor/EnterExit14/EnterExit14.sh             
baseservices/threading/generics/Monitor/TryEnter01/TryEnter01.sh               
baseservices/threading/generics/Monitor/TryEnter03/TryEnter03.sh               
baseservices/threading/generics/Monitor/TryEnter04/TryEnter04.sh               
baseservices/threading/generics/Monitor/TryEnter05/TryEnter05.sh               
baseservices/threading/generics/Monitor/TryEnter06/TryEnter06.sh               
baseservices/threading/generics/TimerCallback/thread01/thread01.sh             
baseservices/threading/generics/TimerCallback/thread02/thread02.sh             
baseservices/threading/generics/TimerCallback/thread03/thread03.sh             
baseservices/threading/generics/TimerCallback/thread04/thread04.sh             
baseservices/threading/generics/TimerCallback/thread05/thread05.sh             
baseservices/threading/generics/TimerCallback/thread06/thread06.sh                                                                                            
baseservices/threading/generics/TimerCallback/thread07/thread07.sh             
baseservices/threading/generics/TimerCallback/thread08/thread08.sh             
baseservices/threading/generics/TimerCallback/thread09/thread09.sh             
baseservices/threading/generics/TimerCallback/thread10/thread10.sh                                                                                            
baseservices/threading/generics/TimerCallback/thread11/thread11.sh             
baseservices/threading/generics/TimerCallback/thread12/thread12.sh                                                                                            
baseservices/threading/generics/TimerCallback/thread13/thread13.sh                                                                                            
baseservices/threading/generics/TimerCallback/thread14/thread14.sh                                                                                            
baseservices/threading/generics/TimerCallback/thread15/thread15.sh             
baseservices/threading/generics/TimerCallback/thread16/thread16.sh             
baseservices/threading/generics/TimerCallback/thread17/thread17.sh             
baseservices/threading/generics/TimerCallback/thread18/thread18.sh             
baseservices/threading/generics/TimerCallback/thread19/thread19.sh             
baseservices/threading/generics/TimerCallback/thread20/thread20.sh             
baseservices/threading/generics/TimerCallback/thread21/thread21.sh             
baseservices/threading/generics/TimerCallback/thread22/thread22.sh             
baseservices/threading/generics/TimerCallback/thread23/thread23.sh             
baseservices/threading/generics/TimerCallback/thread24/thread24.sh             
baseservices/threading/generics/TimerCallback/thread25/thread25.sh             
baseservices/threading/generics/TimerCallback/thread26/thread26.sh             
baseservices/threading/generics/TimerCallback/thread27/thread27.sh             
baseservices/threading/generics/TimerCallback/thread28/thread28.sh             
baseservices/threading/generics/TimerCallback/thread29/thread29.sh             
baseservices/threading/generics/TimerCallback/thread30/thread30.sh             
baseservices/threading/generics/TimerCallback/tighttimercallback/tighttimercallback.sh 
baseservices/threading/generics/WaitCallback/thread01/thread01.sh              
baseservices/threading/generics/WaitCallback/thread02/thread02.sh              
baseservices/threading/generics/WaitCallback/thread03/thread03.sh              
baseservices/threading/generics/WaitCallback/thread04/thread04.sh              
baseservices/threading/generics/WaitCallback/thread05/thread05.sh              
baseservices/threading/generics/WaitCallback/thread06/thread06.sh              
baseservices/threading/generics/WaitCallback/thread07/thread07.sh              
baseservices/threading/generics/WaitCallback/thread07/thread07.sh 
baseservices/threading/generics/WaitCallback/thread08/thread08.sh 
baseservices/threading/generics/WaitCallback/thread09/thread09.sh 
baseservices/threading/generics/WaitCallback/thread10/thread10.sh 
baseservices/threading/generics/WaitCallback/thread11/thread11.sh 
baseservices/threading/generics/WaitCallback/thread12/thread12.sh 
baseservices/threading/generics/WaitCallback/thread13/thread13.sh 
baseservices/threading/generics/WaitCallback/thread14/thread14.sh 
baseservices/threading/generics/WaitCallback/thread15/thread15.sh 
baseservices/threading/generics/WaitCallback/thread16/thread16.sh 
baseservices/threading/generics/WaitCallback/thread17/thread17.sh 
baseservices/threading/generics/WaitCallback/thread18/thread18.sh 
baseservices/threading/generics/WaitCallback/thread19/thread19.sh 
baseservices/threading/generics/WaitCallback/thread20/thread20.sh 
baseservices/threading/generics/WaitCallback/thread21/thread21.sh 
baseservices/threading/generics/WaitCallback/thread22/thread22.sh 
baseservices/threading/generics/WaitCallback/thread23/thread23.sh 
baseservices/threading/generics/WaitCallback/thread24/thread24.sh 
baseservices/threading/generics/WaitCallback/thread25/thread25.sh 
baseservices/threading/generics/WaitCallback/thread26/thread26.sh 
baseservices/threading/generics/WaitCallback/thread27/thread27.sh 
baseservices/threading/generics/WaitCallback/thread28/thread28.sh 
baseservices/threading/generics/WaitCallback/thread29/thread29.sh 
baseservices/threading/generics/WaitCallback/thread30/thread30.sh 
baseservices/threading/monitor/enter/monitorenter/monitorenter.sh 
baseservices/threading/monitor/isentered/monitorisentered/monitorisentered.sh 
baseservices/threading/regressions/13662/13662-a/13662-a.sh 
baseservices/threading/regressions/13662/13662-b/13662-b.sh 
baseservices/threading/regressions/13662/13662-simple/13662-simple.sh 
baseservices/threading/regressions/30032/30032/30032.sh 
baseservices/threading/regressions/beta2/437017/437017.sh 
baseservices/threading/regressions/devdiv489437/test489437/test489437.sh 
baseservices/threading/threadpool/unregister/regression_749068/regression_749068.sh 
baseservices/threading/threadstatic/threadstatic07/threadstatic07.sh 
GC/API/GC/GetTotalAllocatedBytes/GetTotalAllocatedBytes.sh 
JIT/opt/Devirtualization/box2/box2.sh 
JIT/Performance/CodeQuality/BenchmarksGame/binarytrees/binarytrees-5/binarytrees-5.sh 
JIT/Performance/CodeQuality/BenchmarksGame/fasta/fasta-1/fasta-1.sh 
JIT/Performance/CodeQuality/BenchmarksGame/k-nucleotide/k-nucleotide-9/k-nucleotide-9.sh 
JIT/Performance/CodeQuality/BenchmarksGame/mandelbrot/mandelbrot-7/mandelbrot-7.sh 
JIT/Performance/CodeQuality/BenchmarksGame/spectralnorm/spectralnorm-3/spectralnorm-3.sh 
JIT/Performance/CodeQuality/Roslyn/CscBench/CscBench.sh 
JIT/Performance/CodeQuality/SIMD/RayTracer/RayTracer/RayTracer.sh 
JIT/Regression/CLR-x86-JIT/V2.0-Beta2/b425314/b425314/b425314.sh 
JIT/Regression/JitBlue/GitHub_19361/GitHub_19361/GitHub_19361.sh 
JIT/Regression/JitBlue/GitHub_27924/GitHub_27924/GitHub_27924.sh 
profiler/eventpipe/eventpipe/eventpipe.sh 
profiler/eventpipe/eventpipe_readevents/eventpipe_readevents.sh 
profiler/gc/gc/gc.sh 
profiler/gc/gcbasic/gcbasic.sh 
profiler/unittest/getappdomainstaticaddress/getappdomainstaticaddress.sh 
profiler/unittest/metadatagetdispenser/metadatagetdispenser.sh 
tracing/eventpipe/gcdump/gcdump/gcdump.sh 
```

/cc @dotnet/dotnet-diag

### PR DESCRIPTION
**Pull request by sywhang**
Fix #48233 

@sdmaclea would it be possible for you to run the failing tests on this branch? I don't have an M1 mac at hand to test this.

cc @noahfalk

### CODE REVIEW

**Comment by sdmaclea** [comment]
Running tests now.  It might take a while (I am running gcstress in another terminal).

**Comment by sdmaclea** [comment]
This fixed most of the issues.  There are three failing tracing tests still.  Do these need to be disabled for MacOS?.

```
tracing/tracevalidation/tracelogging/tracelogging/tracelogging.sh

Unhandled exception. System.InvalidOperationException: Reflection failed.
   at Tracing.Tests.Common.TraceConfiguration..ctor(String outputFile, UInt32 circularBufferMB) in /Users/stmaclea/git/runtime/src/tests/tracing/common/TraceC
onfiguration.cs:line 30                                                                                                                                       
   at Tracing.Tests.Common.EventSourceTestSuite.GenerateConfiguration() in /Users/stmaclea/git/runtime/src/tests/tracing/tracevalidation/tracelogging/EventSou
rceTest.cs:line 107                                                            
   at Tracing.Tests.Common.EventSourceTestSuite.RunTests() in /Users/stmaclea/git/runtime/src/tests/tracing/tracevalidation/tracelogging/EventSourceTest.cs:li
ne 53                                                                          
   at Tracing.Tests.TraceLogging.Main(String[] args) in /Users/stmaclea/git/runtime/src/tests/tracing/tracevalidation/tracelogging/TraceLogging.cs:line 123   
/Users/stmaclea/git/runtime/artifacts/tests/coreclr/OSX.arm64.Checked/tracing/tracevalidation/tracelogging/tracelogging/tracelogging.sh: line 395:  9522 Abort trap: 6           $LAUNCHER $ExePath "${CLRTestExecutionArguments[@]}" 
```
```
tracing/tracevalidation/rundown/rundown/rundown.sh                                 
                                                                                                                                                              
Unhandled exception. System.TypeInitializationException: The type initializer for 'Tracing.Tests.Common.TraceControl' threw an exception.                     
 ---> System.InvalidOperationException: Reflection failed.                                                                                                    
   at Tracing.Tests.Common.TraceControl..cctor() in /Users/stmaclea/git/runtime/src/tests/tracing/common/TraceControl.cs:line 82                              
   --- End of inner exception stack trace ---                                  
   at Tracing.Tests.TraceValidationRundown.Main(String[] args) in /Users/stmaclea/git/runtime/src/tests/tracing/tracevalidation/rundown/Rundown.cs:line 30    
/Users/stmaclea/git/runtime/artifacts/tests/coreclr/OSX.arm64.Checked/tracing/tracevalidation/rundown/rundown/rundown.sh: line 402:  9525 Abort trap: 6       
    $LAUNCHER $ExePath "${CLRTestExecutionArguments[@]}"              
```
```
tracing/tracevalidation/jittingstarted/JittingStarted/JittingStarted.sh

Unhandled exception. System.TypeInitializationException: The type initializer for 'Tracing.Tests.Common.TraceControl' threw an exception.                     
 ---> System.InvalidOperationException: Reflection failed.                                                                                                    
   at Tracing.Tests.Common.TraceControl..cctor() in /Users/stmaclea/git/runtime/src/tests/tracing/common/TraceControl.cs:line 82                              
   --- End of inner exception stack trace ---                                                                                                                 
   at Tracing.Tests.Common.TraceControl.EnableDefault(String outputFile) in /Users/stmaclea/git/runtime/src/tests/tracing/common/TraceControl.cs:line 21      
   at Tracing.Tests.TraceValidationJittingStarted.Main(String[] args) in /Users/stmaclea/git/runtime/src/tests/tracing/tracevalidation/jittingstarted/JittingS
tarted.cs:line 26                 
/Users/stmaclea/git/runtime/artifacts/tests/coreclr/OSX.arm64.Checked/tracing/tracevalidation/jittingstarted/JittingStarted/JittingStarted.sh: line 402:  9518
 Abort trap: 6           $LAUNCHER $ExePath "${CLRTestExecutionArguments[@]}"                                                                                 
```

**Comment by sywhang** [comment]
@sdmaclea Those tests were removed from https://github.com/dotnet/runtime/pull/48053. To give you a little bit more context, those tests depended on a removed internal API that was provided for an internal customer but we moved that customer off that, so all the tests that were testing that feature were removed with it.

**Comment by sdmaclea** [comment]
Interesting.  I saw the failures and wondered if they might be dependent on the flag you set, so I rebuilt the tests.  However I didn't delete the originally compiled tests, so I think they must have just ended up in the XUnit Wrappers anyway.

File: src/coreclr/clrdefinitions.cmake [hunk 1]
```diff
@@ -120,9 +120,9 @@ endif(FEATURE_GDBJIT_LANGID_CS)
 if(FEATURE_GDBJIT_SYMTAB)
     add_definitions(-DFEATURE_GDBJIT_SYMTAB)
 endif(FEATURE_GDBJIT_SYMTAB)
-if(CLR_CMAKE_TARGET_UNIX)
+if(CLR_CMAKE_TARGET_LINUX)
```
Comment (sdmaclea) [inline src/coreclr/clrdefinitions.cmake:123]: I wonder if this should actually be controlled by the cmake `FeatureXplatEventSource` property set  above.  That would keep them consistent.

**Review by sdmaclea — APPROVED** [review]

**Pull request merged**